pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -513,21 +513,22 @@ func TestExpectationsOnRecreate(t *testing.T) {
 	waitForQueueLength := func(expected int, msg string) {
 		t.Helper()
 		i := 0
-		err = wait.PollImmediate(100*time.Millisecond, informerSyncTimeout, func() (bool, error) {
-			current := dsc.queue.Len()
-			switch {
-			case current == expected:
-				return true, nil
-			case current > expected:
-				return false, fmt.Errorf("queue length %d exceeded expected length %d", current, expected)
-			default:
-				i++
-				if i > 1 {
-					t.Logf("Waiting for queue to have %d item, currently has: %d", expected, current)
+		err = wait.PollUntilContextTimeout(context.Background(), 100*time.Millisecond, informerSyncTimeout, true,
```

Comment:
would it make sense to use `ctx` here instead of `context.Background()`?

Reply:
get it